packages/@react-stately/data/src/useTreeData.ts

```diff
@@ -322,6 +322,16 @@ export function useTreeData<T extends object>(options: TreeOptions<T>): TreeData
           parentKey: toParentKey
         };
 
+        // If parentKey is null, insert into the root.
+        if (toParentKey == null) {
```

Comment:
Do we need to support an item changing position within the root, like it's the last item out of three items and we want it to be the first item in the root? `move({name: 'Adam', parentKey: null}, null, 0)` from this tree `[{'David'}, {'John'}, {'Adam'}]`